backend/README.md

```diff
@@ -134,4 +137,11 @@ Below are some exceptions and additions to the guidelines.
 ### Named return values and Naked returns
 
 - Prefer to not use named return values. Return values should be obvious from the context, their types, and finally from the function/method comment.
-- Don't use naked returns. They make the code harder to read and understand.
\ No newline at end of file
+- Don't use naked returns. They make the code harder to read and understand.
+
+### Updating type and query definitions
+During the development process, it is sometimes required to update type and queries, e.g., for `Process`, `Instance`, `Timer`, etc. All the changes must be made in `graph/schema.graphqls`:
+- Open the file and navigate to the type/query you want to update.
+- Update the type / query with correct fields. When you add new queries, you might also want to remove `@goField(forceResolver: true)` accordingly.
+- Under `/backend`, run `./run_gqlgen.sh` to update query in automatically generated files (`generated.go`, `models_gen.go`, `schema.resolvers.go`). DO NOT manually update those files.
+- Run the backend again `./run_api.sh` and open the playground to test the new changes.
```

Comment:
I'm going to leave this PR for someone else to review, but just going to comment something:

> When you add new queries, you might also want to remove `@goField(forceResolver: true)` accordingly.

The resolver is forced to allow loading only instances which belong to a certain process, as it allows inspecting the incoming query for a process key. So these shouldn't be removed.

> Under `/backend`, run `./run_gqlgen.sh` to update query in automatically generated files (`generated.go`, `models_gen.go`, `schema.resolvers.go`). DO NOT manually update those files.

`schema.resolvers.go` is partly generated, but generated functions are fine to edit as these make the query itself. FYI, there is a warning (a comment and extra code) in that file that should be resolved before merging.